pkg/parser/parser.go

```diff
@@ -24,10 +27,38 @@ func executeBootstrapTemplate() (string, error) {
 // this function will eventually take a pointer to the bootstrap contract struct.
 // it will then template out the variables into the final bootstrap trigger script.
 func Parse() {
-	triggerBootstrapScript, err := executeBootstrapTemplate()
+	inputJson, err := json.Marshal(getBaseTemplate())
 	if err != nil {
-		log.Printf("Failed to execute the template: %v", err)
+		log.Printf("Failed to Marshal the nbcontractv1 to json: %v", err)
 	}
+	log.Println("Input Json: ")
+	log.Println(string(inputJson))
+
+	//inputJson above will be provided by bootstrappers. We are using getBaseTemplate() for dev/test purpose for now.
+	//We can further move it to other file for unit tests later.
 
+	inputContract := nbcontractv1.Configuration{}
+	json.Unmarshal(inputJson, &inputContract)
+	triggerBootstrapScript, err := executeBootstrapTemplate(&inputContract)
+	if err != nil {
+		log.Printf("Failed to execute the template: %v", err)
+	}
+	log.Println("output env vars:")
 	log.Println(triggerBootstrapScript)
 }
+
+func getBaseTemplate() *nbcontractv1.Configuration {
```

Comment:
can we put this in a different file like how ab e2e does it? https://github.com/Azure/AgentBaker/blob/master/e2e/template.go

also, lets try to use as much of the actual variable values as we can, i.e what we expect from the bootstrapper.

eg, instead of test k8s version, lets use an actual k8s version, similarly for URLs, you can refer to template.go that I linked.

this will make it easier to test once we are ready

Reply:
Sounds good.
Here are what I could find from the template.go. 

PROVISION_OUTPUT="/var/log/azure/cluster-provision-cse-output.log"
REPO_DEPOT_ENDPOINT=""
MOBY_VERSION=""
TENANT_ID=""
KUBERNETES_VERSION="1.26.0"
HYPERKUBE_URL="mcr.microsoft.com/oss/kubernetes/"
KUBE_BINARY_URL=""
CUSTOM_KUBE_BINARY_URL="https://acs-mirror.azureedge.net/kubernetes/v1.26.0/binaries/kubernetes-node-linux-amd64.tar.gz"
KUBEPROXY_URL="